packages/block-library/src/navigation-menu-item/edit.js

```diff
@@ -1,93 +1,131 @@
 	setAttributes,
 } ) {
 	const plainTextRef = useRef( null );
-	const onEditLableClicked = useCallback(
-		( onClose ) => () => {
-			onClose();
-			invoke( plainTextRef, [ 'current', 'textarea', 'focus' ] );
-		},
-		[ plainTextRef ]
-	);
+	const [ isLinkOpen, setIsLinkOpen ] = useState( false );
+	const [ isEditingLink, setIsEditingLink ] = useState( false );
+	const [ urlInput, setUrlInput ] = useState( null );
+
+	const inputValue = urlInput !== null ? urlInput : url;
+
+	const onKeyDown = ( event ) => {
+		if ( [ LEFT, DOWN, RIGHT, UP, BACKSPACE, ENTER ].indexOf( event.keyCode ) > -1 ) {
+			// Stop the key event from propagating up to ObserveTyping.startTypingInTextField.
+			event.stopPropagation();
+		}
+	};
+
+	const closeURLPopover = () => {
+		setIsEditingLink( false );
+		setUrlInput( null );
+		setIsLinkOpen( false );
+	};
+
+	const autocompleteRef = useRef( null );
+
+	const onFocusOutside = ( event ) => {
+		const autocompleteElement = autocompleteRef.current;
+		if ( autocompleteElement && autocompleteElement.contains( event.target ) ) {
+			return;
+		}
+		closeURLPopover();
+	};
+
+	const { label, url } = attributes;
 	let content;
 	if ( isSelected ) {
 		content = (
 			<div className="wp-block-navigation-menu-item__edit-container">
 				<PlainText
 					ref={ plainTextRef }
 					className="wp-block-navigation-menu-item__field"
-					value={ attributes.label }
-					onChange={ ( label ) => setAttributes( { label } ) }
+					value={ label }
+					onChange={ ( labelValue ) => setAttributes( { label: labelValue } ) }
 					aria-label={ __( 'Navigation Label' ) }
 					maxRows={ 1 }
 				/>
-				<Dropdown
-					contentClassName="wp-block-navigation-menu-item__dropdown-content"
-					position="bottom left"
-					popoverProps={ POPOVER_PROPS }
-					renderToggle={ ( { isOpen, onToggle } ) => (
-						<IconButton
-							icon={ isOpen ? 'arrow-up-alt2' : 'arrow-down-alt2' }
-							label={ __( 'More options' ) }
-							onClick={ onToggle }
-							aria-expanded={ isOpen }
-						/>
-					) }
-					renderContent={ ( { onClose } ) => (
-						<MenuItemActions
-							clientId={ clientId }
-							destination={ attributes.destination }
-							onEditLableClicked={ onEditLableClicked( onClose ) }
-						/>
-					) }
-				/>
 			</div>
 		);
 	} else {
-		content = attributes.label;
+		content = label;
 	}
 	return (
 		<Fragment>
+			<BlockControls>
+				<Toolbar>
+					<ToolbarButton
+						name="link"
+						icon="admin-links"
+						title={ __( 'Link' ) }
+						onClick={ () => setIsLinkOpen( ! isLinkOpen ) }
+					/>
+					{ isLinkOpen &&
+					<>
+						<URLPopover
+							className="wp-block-navigation-menu-item__inline-link-input"
+							onClose={ closeURLPopover }
+							onFocusOutside={ onFocusOutside }
+						>
+							{ ( ! url || isEditingLink ) &&
+							<URLPopover.LinkEditor
+								value={ inputValue }
+								onChangeInputValue={ setUrlInput }
+								onKeyPress={ ( event ) => event.stopPropagation() }
```

Comment:
As I mentioned in [this comment](https://github.com/WordPress/gutenberg/pull/17846#pullrequestreview-305454563), I think `onKeyPress` event is [not already handled](https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/src/components/url-input/index.js#L24-L39) by the `<URLPopover.LinkEditor />`, and just in case [this commit](https://github.com/WordPress/gutenberg/pull/18062/commits/ca73107cdb9aa4515df7ad25437cd7752e92f48a) takes over of this.

Reply:
The keypress event handler passed into LinkEditor is [attached to the form element](https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/src/components/url-popover/link-editor.js#L32), not to URLInput. Your commit shouldn't have any effect on this.

Reply:
Sorry I got confused last night. Thanks @tellthemachines and sorry again to bring unnecessary noise here.

Reply:
No worries, always best to comment if you're not sure!